src/libsyntax/ext/build.rs

```diff
@@ -844,7 +844,7 @@ impl<'a> AstBuilder for ExtCtxt<'a> {
     }
     fn pat_enum(&self, span: Span, path: ast::Path, subpats: Vec<P<ast::Pat>>) -> P<ast::Pat> {
         let pat = if subpats.is_empty() {
-            PatKind::Path(None, path)
+            PatKind::Struct(path, Vec::new(), false)
```

Comment:
This is a major change cc @mcarton and @dtolnay 

`PatKind::Struct` is now used instead of `PatKind::Path` for the empty enum case.


Is this 100% necessary, though?